src/components/client-request-cache.ts

```diff
@@ -17,58 +17,59 @@ limitations under the License.
 /**
  * Caches requests in memory and handles expiring them.
  */
-class ClientRequestCache {
+export class ClientRequestCache<T> {
+    private requestContent = new Map<string, {ts: number, content: T}>();
     /**
-     * @param ttl {Number} How old a result can be before it gets expired.
-     * @param size {Number} How many results to store before we trim.
+     * @param ttl How old a result can be before it gets expired.
+     * @param size How many results to store before we trim.
      * @param requestFunc The function to use on cache miss.
      */
-    constructor (ttl, size, requestFunc) {
+    constructor (private readonly ttl: number,
+        private readonly maxSize: number,
+        private readonly requestFunc: (...args: any[]) => Promise<T>) {
```

Comment:
If that doesn't break TypeScripts analysis:
```suggestion
        private readonly requestFunc: (...args: unknown[]) => Promise<T>) {
```